openedx/adg/lms/applications/admin.py

```diff
@@ -72,3 +81,284 @@ class BusinessLineAdmin(admin.ModelAdmin):
+        Return the html for course names of all prereq courses and applicant's respective scores in those courses
+        """
+        html_for_score = '<p>{course_name}: <b>{course_percentage}%</b></p>'
+        final_html = ''
+        prereq_scores = obj.prereq_course_scores
+        for prereq_score in prereq_scores:
+            final_html += html_for_score.format(
+                course_name=prereq_score.course_name,
+                course_percentage=prereq_score.course_percentage
+            )
+
+        return format_html(final_html)
+
+    def get_fieldsets(self, request, obj=None):
+        basic_info_fields = ['email', 'location']
+        if obj.linkedin_url:
+            basic_info_fields.append('linkedIn_profile')
+
+        fieldsets = [
+            (None, {
+                'fields': tuple(basic_info_fields)
+            }),
+            ('APPLICANT INFORMATION', {
+                'fields': (
+                        'is_saudi_national', 'gender', 'phone_number', 'date_of_birth', 'organization', 'applying_to'
+                ),
+            }),
+        ]
+
+        file_fields = []
+        fieldset_for_files = None
+
+        if obj.is_cover_letter_provided and obj.resume:
+            fieldset_for_files = RESUME_AND_COVER_LETTER
+            file_fields.append('resume')
+            if obj.cover_letter_file:
+                file_fields.append('cover_letter_file')
+        elif obj.resume:
+            fieldset_for_files = RESUME_ONLY
+            file_fields.append('resume')
+        elif obj.is_cover_letter_provided:
+            fieldset_for_files = COVER_LETTER_ONLY
+            if obj.cover_letter_file:
+                file_fields.append('cover_letter_file')
+
+        if fieldset_for_files:
+            if obj.resume:
+                if can_display_file(obj.resume):
+                    file_fields.append('resume_display')
+
+            if obj.cover_letter_file:
+                if can_display_file(obj.cover_letter_file):
+                    file_fields.append('cover_letter_display')
+            elif obj.cover_letter_text:
+                file_fields.append('cover_letter_text')
+
+            fieldsets.append((fieldset_for_files, {'fields': tuple(file_fields)}))
+
+        fieldset_for_scores = ('SCORES', {'fields': ('prerequisites', )})
+
+        fieldsets.append(fieldset_for_scores)
+
+        return tuple(fieldsets)
+
+    form = UserApplicationAdminForm
+
+    readonly_fields = (
+        'email',
+        'location',
+        'linkedIn_profile',
+        'is_saudi_national',
+        'gender',
+        'phone_number',
+        'date_of_birth',
+        'organization',
+        'applying_to',
+        'resume',
+        'cover_letter_file',
+        'cover_letter_display',
+        'resume_display',
+        'cover_letter_text',
+        'prerequisites'
+    )
+
+    inlines = [
+        EducationInline, WorkExperienceInline
+    ]
+
+    def get_formsets_with_inlines(self, request, obj=None):
```

Comment:
Let's please add docstrings, and if a method is an override, then mention the reason to override.